edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet.Docker/CombinedEdgeletConfigProvider.cs

```diff
@@ -139,5 +143,44 @@ void MountSockets(IModule module, CreateContainerParameters createOptions)
                 SetMountOptions(createOptions, managementUri);
             }
         }
+
+        void SetCapabilities(CreateContainerParameters createOptions)
+        {
+            // These capabilities are provided by default and can be used to gain root access:
+            // https://labs.f-secure.com/blog/helping-root-out-of-the-container/
+            HashSet<string> capabilitiesToRemove = new HashSet<string> { "CAP_CHOWN", "CAP_SETUID" };
+
+            // If customer manually adds the capabilites, don't drop them.
+            if (createOptions.HostConfig?.CapAdd != null)
```

Comment:
Couldn't all this be simplified to something like:

```
createOptions.HostConfig.CapDrop =
    new List<string> { "CAP_CHOWN", "CAP_SETUID" }.except(createOptions.HostConfig.CapAdd);
```

(This is just from a quick glance at it, please correct me if I'm wrong)